### PR TITLE
add homework 12

### PR DESCRIPTION
**Pull request by volodymyrrusin**
Please check my homework for lesson 12

### CODE REVIEW

File: 12/homework.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
+from datetime import datetime
+import json
+
+
+# TASK 2: Create decorator with saving text into a txt file
+def my_decorator(func):
+    def inner(*args, **kwargs):
+        text = f'Function name is {func.__name__}, time of execution is {datetime.now()}\n'
+        try:
+            with open('my_deco.txt'):
```
Comment (klymenok) [inline 12/homework.py:10]: Why do you open this file twice here?

File: 12/homework.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        return result
+    return inner
+
+
+# TASK 3: Create custom exception with saving text into a txt file
+class MyCustomException(Exception):
+    def __init__(self, message='Custom exception is occurred'):
+        text = f'{message}, date of occurrence is {datetime.now()}\n'
+        try:
+            with open('my_exception.txt'):
```
Comment (klymenok) [inline 12/homework.py:26]: Why do you open this file twice here?

File: 12/homework.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+                    file.write(text)
+        except IOError:
+            with open('my_exception.txt', 'w') as file:
+                file.write(text)
+        super().__init__(message)
+
+
+# TASK 1: Create saving phone book into a json file
+try:
+    with open('phone_book.json'):
```
Comment (klymenok) [inline 12/homework.py:37]: Why do you open this file twice here?

**Review by klymenok — CHANGES_REQUESTED** [review]

File: 12/homework.py (hunk 3 above)
Comment (volodymyrrusin) [reply 12/homework.py:37]: For all the tasks I thought that with first "open" I would check if the file already exist and then open it in appropriate mode.
But now I understood that It works without first "open" as well. Shoud I make these changes (delete these lines of code)?
Also I see that I made mistake in line 14 - didn't change mode "a" into "w".

Comment (klymenok) [reply 12/homework.py:37]: Yes, remove extra `open`, please. You can use `w+`, `a+` to create a file if it does not exist before writing

**Review by klymenok — APPROVED** [review]
Looks good!

**Pull request merged**